### ISSUE TITLE
Add option to not use Trash bin

### ISSUE THREAD

**Issue by danielb987** [desktop/desktop#10445]
I never ever use the trash bin on either Windows or Linux. I always delete files directly. But when GitHub Desktop discards changes, it moves the old file to the trash bin and therefore forcing me to empty the trash bin.

There is two problems with this:
* Since I never use the trash bin, I don't realize that the trash bin is filled with a lot of files
* Sometimes, on Linux, the trash bin makes the computer slow down if it contains a lot of files

I would like an option to never use the trash bin, so that "discard changes" does exactly that, discards them without the possibility to get them back.

**Comment by tierninho** [comment]
@danielb987 Thank you for the suggestion. I believe our intention is not to permanently delete the files but to actually just put them in the trash. I would have to say it is better to error on the side of caution and the trash can idea was the right call. 

We have had multiple issues where files were recovered because of this. I'd hate to not be able to help those users.

In the screenshots, you can see both `Discard Files` and `Remove Repo` mention the Trash.

![Screen Shot 2020-08-25 at 2 11 23 PM](https://user-images.githubusercontent.com/14828183/91240242-20c28000-e6dd-11ea-9aa8-b4dac28a3f13.png)
![Screen Shot 2020-08-25 at 2 11 47 PM](https://user-images.githubusercontent.com/14828183/91240246-21f3ad00-e6dd-11ea-87e6-3df821f82178.png)

We welcome any comments for users about changing this, but for now the issue will be closed.

**Comment by danielb987** [comment]
@tierninho 
Thanks for your comment, but it seems that you haven't understand my suggestion.

I want the **option** to get the files deleted at once instead of moving them to the trash. I agree that the default should be to move the files to the trash, but I want the possibility to get the files deleted at once.

**Comment by tierninho** [comment]
Ah, gotcha. Will reopen so others can comment. 👍

**Comment by danielb987** [comment]
Thanks!

**Comment by aryabchenko** [comment]
I also want such an option, cause it takes forever on my system to discard hundreds of files (cause they are moved to trash)

**Comment by MichaIng** [comment]
Same here, discarding changes on a large repository takes much too long when a simple `git reset --hard` would do it. I would even suggest to make this the default: Git doesn't have a trash bin. A simple checkbox like shown above when removing a repository should however be pretty fine.

**Comment by AdityaSavara** [comment]
I agree with the suggestion here. Without seeing this thread, I had made a very similar suggestion on another thread with a related issue: https://github.com/desktop/desktop/issues/7155

I want to emphasize that I would prefer a popup for us to decide each time rather than something buried in settings. Because I think sending the discards to trash/recycle is not intuitive and I would not have known what was occurring without reading the issues threads. A popup box with the possibility to "remember my choice" would be a good way to put it in settings if github desktop developers feel this should be a setting rather than a popup each time.

**Comment by joyfullservice** [comment]
Just another factor to consider here... There is a potential security risk/vulnerability in moving files to the recycle bin. Let me give a couple of example scenarios...

While reviewing the changes before making a commit, a developer realizes that he inadvertently included a real credit card number in his testing code. He discards the change, then goes back and rewrites the test using test card numbers. What he didn't realize is that this credit card number is now sitting in plain text on his computer hard drive, and could potentially be there for a long period of time.

Another developer discovers that his computer was compromised through a phishing attack. Forensics determines that customer PII was found on his computer even though the developer is sure that he never committed code containing PII. (The customer information was discovered in the recycle bin from some spreadsheets that had been discarded from GitHub Desktop.) This creates a huge legal issue with serious ramifications for the company.

Other potential cases would be non-disclosure agreements or malware components. Essentially, it just seems risky to have no way to turn off the sending of these files to the recycle bin. (For the record, I think it is a great feature, and agree that it should definitely be on by default. I just agree that there should be a way to turn it off in environments dealing with more sensitive data.)

**Comment by tidy-dev** [comment]
Some users may find this system setting on Windows useful. 

![image](https://user-images.githubusercontent.com/75402236/221627196-e5bc8e8d-28a6-4444-837a-794d30d31886.png)

**Comment by MichaIng** [comment]
> Some users may find this system setting on Windows useful.

Of course, but whether one want's GitHub desktop to not move everything to trash when deleting repos or discarding changes, is a very different thing to whether one wants the Windows Explorer to use the trash bin on file deletion.

**Comment by aminya** [comment]
Is this why discarding takes so much time making it unusable?
https://github.com/desktop/desktop/issues/7155

**Comment by aminya** [comment]
Made the pull request #17436, which fixes this issue.

**Comment by danielb987** [comment]
> Made the pull request #17436, which fixes this issue.

Thank you!
